tests/models/UserProfilingModel.spec.ts

```diff
@@ -299,6 +299,41 @@ describe('UserProfilingModel', () => {
                 });
             });
 
+            it('should fetch all actions of a user from the backend only once', async () => {
```

Comment:
Your tests in this file are a bit complex, I think that's normal, however, I think we could improve them by adding some comments to makes them easier to understand at first glance ;)

Reply:
Read the it string :trollface: . (Yeah will add a little something)